core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -1036,22 +1072,64 @@ public long getMaximumValueCount(Solution_ solution) {
                 .orElse(0L);
     }
 
+    public record ProblemScaleTracker(long logBase,
+            MutableLong basicProblemScaleLog,
+            Set<Object> visitedAnchorSet,
+            MutableLong listPinnedValueCount,
+            MutableLong listTotalEntityCount,
+            MutableLong listMovableEntityCount,
+            MutableLong listTotalValueCount) {
+        public void addBasicProblemScale(long count) {
+            basicProblemScaleLog.add(MathUtils.getScaledApproximateLog(MathUtils.LOG_PRECISION, logBase, count));
+        }
+    }
+
     /**
      * Calculates an indication on how big this problem instance is.
-     * This is intentionally very loosely defined for now.
+     * This is approximately the base 10 log of the solution space size.
```

Comment:
Solution space size... Maybe we use "search space size" instead? It's closer to the industry lingo. (Assuming that search space size is what you mean here.)